librz/bin/dwarf/loclists.c

```diff
@@ -0,0 +1,338 @@
+		}
+		break;
+	}
+	}
+	memcpy(out, &entry, sizeof(entry));
+	return true;
+}
+
+void RzBinDwarfLocationListEntry_free(RzBinDwarfLocationListEntry *self) {
+	if (!self) {
+		return;
+	}
+	free(self->range);
+	free(self);
+}
+
+static bool convert_raw(RzBinDwarfLocListTable *self, RzBinDwarfRawLocListEntry *raw, RzBinDwarfLocationListEntry **out) {
+	ut64 mask = self->encoding.address_size == 0 ? ~0ULL : (~0ULL >> (64 - self->encoding.address_size * 8));
+	ut64 tombstone = self->encoding.version <= 4 ? mask - 1
+						     : mask;
+	RzBinDwarfRange *range = NULL;
+	RzBinDwarfBlock *data = NULL;
+	if (raw->is_address_or_offset_pair) {
+		if (self->base_address == tombstone) {
+			OK_None;
+		}
+		range = RZ_NEW0(RzBinDwarfRange);
+		range->begin = raw->address_or_offset_pair.begin;
+		range->end = raw->address_or_offset_pair.end;
+		Range_add_base_address(range, self->base_address, self->encoding.address_size);
+		data = RzBinDwarfBlock_clone(&raw->address_or_offset_pair.data);
+		RET_FALSE_IF_FAIL(data);
+	} else {
+		switch (raw->encoding) {
+		case DW_LLE_end_of_list: break;
+		case DW_LLE_base_address:
+			self->base_address = raw->base_address.addr;
+			OK_None;
+		case DW_LLE_base_addressx:
+			RET_FALSE_IF_FAIL(self->debug_addr);
+			RET_FALSE_IF_FAIL(DebugAddr_get_address(self->debug_addr, &self->base_address,
+				self->encoding.address_size, self->encoding.big_endian,
+				self->base_address, raw->base_addressx.addr));
+			OK_None;
+		case DW_LLE_startx_endx:
+			range = RZ_NEW0(RzBinDwarfRange);
+			range->begin = raw->startx_endx.begin;
+			range->end = raw->startx_endx.end;
+			break;
+		case DW_LLE_startx_length:
+			range = RZ_NEW0(RzBinDwarfRange);
+			range->begin = raw->startx_length.begin;
+			range->end = (raw->startx_length.length + raw->startx_length.begin) & mask;
+			break;
+		case DW_LLE_offset_pair:
+			if (self->base_address == tombstone) {
+				OK_None;
+			}
+			range = RZ_NEW0(RzBinDwarfRange);
+			range->begin = raw->address_or_offset_pair.begin;
+			range->end = raw->address_or_offset_pair.end;
+			Range_add_base_address(range, self->base_address, self->encoding.address_size);
+			data = RzBinDwarfBlock_clone(&raw->address_or_offset_pair.data);
+			RET_FALSE_IF_FAIL(data);
+			break;
+		case DW_LLE_default_location: break;
+		case DW_LLE_start_end:
+			range = RZ_NEW0(RzBinDwarfRange);
+			range->begin = raw->startx_endx.begin;
+			range->end = raw->startx_endx.end;
+			break;
+		case DW_LLE_start_length:
+			range = RZ_NEW0(RzBinDwarfRange);
+			range->begin = raw->startx_length.begin;
+			range->end = (raw->startx_length.length + raw->startx_length.begin) & mask;
+			break;
+		case DW_LLE_GNU_view_pair: break;
+		}
+	}
+
+	if (!range) {
+		return false;
+	}
+	if (range->begin == tombstone) {
+		free(range);
+		RzBinDwarfBlock_free(data);
+		OK_None;
+	}
+	if (range->begin > range->end) {
+		RZ_LOG_WARN("Invalid Address Range (0x%" PFMT64x ",0x%" PFMT64x ")\n", range->begin, range->end);
```

Comment:
this is an error, not a warning.
```suggestion
		RZ_LOG_ERROR("dwarf: invalid address range (0x%" PFMT64x ",0x%" PFMT64x ")\n", range->begin, range->end);
```